### PR TITLE
For #28010 - Update documentation for browser_search_with_ads and browser_search_ad_clicks

### LINKED ISSUE #28010: Update documentation for browser_search_with_ads and browser_search_ad_clicks
After the changes in [A-C #229](https://github.com/mozilla-mobile/firefox-android/pull/229) are landed we will need to update the documentation for [browser_search_with_ads](https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/browser_search_with_ads) and [browser_search_ad_clicks](https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/browser_search_ad_clicks)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-22071)

### PR DESCRIPTION
**Pull request by Alexandru2909**
Update documentation for `browser_search_with_ads` and `browser_search_ad_clicks` to reflect the current state

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #28010

### CODE REVIEW

**Comment by Alexandru2909** [comment]
# Request for changed data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**
- These probes will be used to better model high-level trends in revenue prediction. They will record counts of search engine result pages with ads displayed and their search access points.

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**
- This measure helps Mozilla better understand revenue trends and (potentially) aid future forecasting. It also helps us have confidence on the revenue impact of features.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**
- No alternative methods.

4) **Can current instrumentation answer these questions?**
- Current events only track the record counts of SERP pages with ads.

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. **
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
Records counts of SERP pages with adverts displayed. The key format is `<provider-name>.in-content.[sap\|sap-follow-on\|organic].[code\|none](.[channel])?`. | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/28010
Records clicks of adverts on SERP pages. The key format is `<provider-name>.in-content.[sap\|sap-follow-on\|organic].[code\|none](.[channel])?`. | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/28010

6) **How long will this data be collected?**
- The collection of these metrics will never expire.

7) **What populations will you measure?**
-  All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**
- Glean

9) **Where do you intend to share the results of your analysis?**
-  Only on Glean and with mobile teams.

File: app/metrics.yaml [hunk 1]
```diff
@@ -6229,7 +6229,8 @@ browser.search:
     type: labeled_counter
     description: |
       Records counts of SERP pages with adverts displayed.
-      The key format is `<provider-name>`.
+      The key format is 
+      `<provider-name>.in-content.[sap|sap-follow-on|organic].[code|none](.[channel])?`.
```
Comment (Dexterp37) [inline app/metrics.yaml:6233]: Would you kindly expand a bit on the docs here? For example, please kindly document under which circumstances we can hit `none` (as flagged in [this comment](https://github.com/mozilla-mobile/firefox-android/pull/229#discussion_r1034534519)).

Would be relevant to also flag what `channel` is and under which circumstances it is present or absent.

File: app/metrics.yaml [hunk 2]
```diff
@@ -6253,7 +6254,8 @@ browser.search:
     type: labeled_counter
     description: |
       Records clicks of adverts on SERP pages.
-      The key format is `<provider-name>`.
+      The key format is 
+      `<provider-name>.in-content.[sap|sap-follow-on|organic].[code|none](.[channel])?`.
```
Comment (Dexterp37) [inline app/metrics.yaml:6258]: As above.

**Review by Dexterp37 — CHANGES_REQUESTED** [review]

File: app/metrics.yaml [hunk 3]
```diff
@@ -6229,7 +6229,12 @@ browser.search:
     type: labeled_counter
     description: |
       Records counts of SERP pages with adverts displayed.
-      The key format is `<provider-name>`.
+      The key format is 
+      `<provider-name>.in-content.[sap|sap-follow-on|organic].[code|none](.[channel])?`, where
+      `provider-name` is the name of the provider,
+      `sap|sap-follow-on|organic` is the search access point,
+      `code` is set when the url matches any of the provider's code prefixes,
+      `channel` is set to the url "channel" query parameter.
```
Comment (Dexterp37) [inline app/metrics.yaml:6237]:
```suggestion
      The key format is
      `<provider-name>.in-content.[sap|sap-follow-on|organic].[code|none](.[channel])?`, where:

      * `provider-name` is the name of the provider,
      * `sap|sap-follow-on|organic` is the search access point,
      * `code` is set when the url matches any of the provider's code prefixes,
      * `channel` is set to the url "channel" query parameter.
```

By using this format [already used here](https://github.com/mozilla-mobile/fenix/blob/e79a16f284df297059586ea3e99d67624499c70e/app/metrics.yaml#L104-L110) properties will show up in a list making it [easier to read in the tools](https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/events_performed_search).

File: app/metrics.yaml [hunk 4]
```diff
@@ -6253,7 +6258,12 @@ browser.search:
     type: labeled_counter
     description: |
       Records clicks of adverts on SERP pages.
-      The key format is `<provider-name>`.
+      The key format is 
+      `<provider-name>.in-content.[sap|sap-follow-on|organic].[code|none](.[channel])?`, where
```
Comment (Dexterp37) [inline app/metrics.yaml:6262]: Same as above.

**Review by Dexterp37 — CHANGES_REQUESTED** [review]

**Review by Dexterp37 — APPROVED** [review]

**Comment by Alexandru2909** [comment]
The data review request should also cover for the changes to the fact emitted in https://github.com/mozilla-mobile/firefox-android/pull/229.

**Comment by Alexandru2909** [comment]
> > The data review request should also cover for the changes to the fact emitted in [mozilla-mobile/firefox-android#229](https://github.com/mozilla-mobile/firefox-android/pull/229).
> 
> Thanks! @Alexandru2909 did you flag any data steward on this?

Hi! I think the data review request and `data-review` label are sufficient, but I also talked with @rocketsroger about this.

**Comment by rocketsroger** [comment]
> # Request for changed data collection review form
> 
> **All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**
> 
>     1. **What questions will you answer with this data?**
> 
> 
>     * These probes will be used to better model high-level trends in revenue prediction. They will record counts of search engine result pages with ads displayed and their search access points.
> 
> 
>     2. **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**
> 
> 
>     * This measure helps Mozilla better understand revenue trends and (potentially) aid future forecasting. It also helps us have confidence on the revenue impact of features.
> 
> 
>     3. **What alternative methods did you consider to answer these questions? Why were they not sufficient?**
> 
> 
>     * No alternative methods.
> 
> 
>     4. **Can current instrumentation answer these questions?**
> 
> 
>     * Current events only track the record counts of SERP pages with ads.
> 
> 
>     5. **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.    **
>        Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Records counts of SERP pages with adverts displayed. The key format is `<provider-name>.in-content.[sap|sap-follow-on|organic].[code|none](.[channel])?`. 	Category 2 - interaction data 	#28010
> Records clicks of adverts on SERP pages. The key format is `<provider-name>.in-content.[sap|sap-follow-on|organic].[code|none](.[channel])?`. 	Category 2 - interaction data 	#28010
> 
>     6. **How long will this data be collected?**
> 
> 
>     * The collection of these metrics will never expire.
> 
> 
>     7. **What populations will you measure?**
> 
> 
>     * All release, beta, and nightly users with telemetry enabled.
> 
> 
>     8. **Please provide a general description of how you will analyze this data.**
> 
> 
>     * Glean
> 
> 
>     9. **Where do you intend to share the results of your analysis?**
> 
> 
>     * Only on Glean and with mobile teams.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

The data collection will remain permanent with kbrosnan@mozilla.com monitoring

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Review by rocketsroger — APPROVED** [review]

**Pull request merged**